crowdsourcehinter/static/js/src/crowdsourcehinter.js

```diff
@@ -0,0 +1,234 @@
+function CrowdsourceHinter(runtime, element, data){
+    //switching units back to a previous question will make a second hinter start up.
+    //executeHinter is used to disable the hinter after switching units in an edX course.
+    var executeHinter = true;
+    $(".crowdsourcehinter_block", element).hide();
+
+    if(executeHinter){
+    var isShowingHintFeedback = false;
+    var voted = false;
+    $(".csh_HintsToUse", element).text("");
+
+    function stopScript(){
+    //This function is used to prevent a particular instance of the hinter from acting after
+    //switching between edX course's units. 
+        executeHinter = false;
+    }
+    Logger.listen('seq_next', null, stopScript);
+    Logger.listen('seq_prev', null, stopScript);
+    Logger.listen('seq_goto', null, stopScript);
```

Comment:
There's not, right now, a clean way to do this in XBlock, but I agree that it would be useful. Off the cuff, I think I'd lean towards some notion of XBlock lifecycle events that are fired by the frontend runtime when XBlocks are loaded (we already have an '.xblock-initialized' event, but it's fired on the Xblock div, and doesn't have any associated data), and then also when they are unloaded from the page.